common/tools/Cargo.toml

```diff
@@ -0,0 +1,10 @@
+[package]
+name = "tools"
+version = "0.1.0"
+authors = ["Libra Association <opensource@libra.org>"]
+license = "Apache-2.0"
+publish = false
+edition = "2018"
+
+[dependencies]
+uuid = { version = "0.7", features = [ "v4" ] }
```

Comment:
I'm going to be annoying, but why adding this new dependency :( it also looks unnecessary large.

if this is to get a random string, I think it's hacky. Using rand is better, and this is actually what uuid does: https://github.com/uuid-rs/uuid/blob/master/src/v4.rs#L26

Reply:
It's also a guarantee that this is a random string that's a valid file / directory name under any system.

If you look closely at the Cargo lock, this PR is strictly reducing the amount of imports: uuid was already imported through mktemp (which this PR expunges).

Reply:
It's also a guarantee that this is a random string that's a valid file / directory name under any system.

If you look closely at the Cargo lock, this PR is strictly reducing the amount of imports: uuid was already imported through mktemp (which this PR expunges).

Reply:
Also, here's the `Cargo.toml`:
https://docs.rs/crate/uuid/0.7.4/source/Cargo.toml
the v4 feature is made to be minimalistic.

Reply:
Since we already use the `hex` crate pretty extensively, you could also format the temp paths as hex.

```rust
let mut rng = rand::thread_rng();
let mut bytes = [0_u8; 16];
rng.fill_bytes(&mut bytes);
// path: String
let path = hex::encode(&bytes);
```

Reply:
I'm not sure I understand what's the benefit as opposed to doing something like

```

// taken from uuid
use rand::RngCore;
let mut rng = rand::thread_rng();
let mut bytes = [0u8; 16];
rng.fill_bytes(&mut bytes);

// convert to hex
let mut dir_uuid = "".to_string();
bytes.iter().for_each(|b| dir_uuid.push_str(&format!("{:02x}", b)));
```

Reply:
I guess I missed the comment of @phlip9, so if we already support the hex crate it's even simpler. (And I guess we can remove it with the code I wrote above at some point :D)

Reply:
Fair enough. less dashes, done!

Reply:
bravo!